R/as_reporting_triangle.R

```diff
@@ -197,41 +181,45 @@ as_reporting_triangle.data.frame <- function(
 #'   to = as.Date("2025-01-05"),
 #'   by = "day"
 #' )
-#' max_delay <- 4
+#'
+#' # max_delay is inferred from matrix dimensions (4 in this case)
 #' rep_tri <- as_reporting_triangle(
 #'   data = rep_tri_mat,
-#'   reference_dates = reference_dates,
-#'   max_delay = max_delay
+#'   reference_dates = reference_dates
 #' )
 #' rep_tri
 as_reporting_triangle.matrix <- function(data,
-                                         max_delay,
-                                         strata = NULL,
                                          delays_unit = "days",
-                                         reference_dates,
+                                         reference_dates = NULL,
                                          ...) {
-  .validate_triangle(
-    triangle = data,
-    max_delay = max_delay,
-    n = nrow(data)
-  )
+  # Use dummy dates starting from 1900 if no reference_dates provided
+  if (is.null(reference_dates)) {
```

Comment:
I am guessing that for downstream functionality to work, we need to enforce that the row.names are dates of some sort? Otherwise I would say we should remove this and allow row.names of `reporting_triangle` objects be either just the row numbers or be dates? 

I think technically someone could have passed in anything for the `as_reporting_triangle.matrix()` function in `main`, we only required it to be a date for `as_reporting_triangle.data.frame()`...